fabfile/run.py

```diff
@@ -203,3 +203,57 @@ def tpch_queries(query_info, connectionURI, pg_version, citus_version, config_fi
         out_val = run(run_string)
         results_file.write(out_val)
         results_file.write('\n')
+
+# Filter (possibly) citus-related valgrind test logs, put under results directory
+# If no error logs exist, then simply put valgrind_success file 
+def filter_put_citus_valgrind_outputs(repo_path):
+    # cd to directory that we performed valgrind(regression) tests
+    os.chdir(os.path.join(repo_path, config.RELATIVE_REGRESS_PATH))
+
+    regression_diffs_exist = False
+
+    # ship regression.diffs (if exists) to result file in order to push to github
+    if os.path.isfile(config.REGRESSION_DIFFS_FILE):
+        run('mv {} {}'.format(config.REGRESSION_DIFFS_FILE, config.RESULTS_DIRECTORY))
+
+        regression_diffs_exist = True
+
+    if os.path.isfile(config.VALGRIND_LOGS_FILE):
+        # get stack trace id that includes calls to citus
+        run('cat {} | grep -i "citus" | awk \'{{ print $1 }}\' | uniq  > trace_ids'.format(config.VALGRIND_LOGS_FILE))
+
+        citus_related_out_exists = True
+
+        if os.path.isfile('trace_ids'):
+            # filter stack traces with stack trace ids that we found above (if any)
+            run('while read line; do grep {} -e $line ; done < trace_ids > {}'.format(
+                config.VALGRIND_LOGS_FILE, 
+                os.path.join(config.RESULTS_DIRECTORY, config.CITUS_RELATED_VALGRIND_LOG_FILE)))
+
+            citus_related_out_exists = True
+
+    # if we have neither regression.diffs nor citus-related valgrind outputs
+    # then just put an empty file named as valgrind_success
+    if not regression_diffs_exist and not citus_related_out_exists:
+        run('touch {}'.format(os.path.join(config.RESULTS_DIRECTORY, 'valgrind_success')))
+
+@task
+@roles('master')
+def valgrind(*args): 
+    'Runs valgrind tests'
+
+    # set citus path variable
+    repo_path = config.settings['repo_path']
```

Comment:
you use 'repo_path' as a hard coded value in 3 different places. We should also add this to config as a variable. I would say that whenever you are adding a hard coded string, consider adding it to as a variable in config.

`config.settings[config.REPO_PATH]`.

Reply:
makes sense, done